src/engine/test/source/metrics/interface_test.cpp

```diff
@@ -0,0 +1,152 @@
+#include <gtest/gtest.h>
+#include <memory>
+
+#include <metrics/metricsManager.hpp>
+#include <json/json.hpp>
+
+using namespace metrics_manager;
+
+class MetricsInterfaceTest : public ::testing::Test
+{
+protected:
+
+    MetricsInterfaceTest() 
+    {
+        m_manager = std::make_shared<MetricsManager>();
+    }
+
+    ~MetricsInterfaceTest() 
+    {
+        
+    }
+
+    void TearDown() override
+    {
+
+    }
+
+    std::shared_ptr<IMetricsManager> m_manager;
+
+};
```

Comment:
Watch out with clang and empty spaces
```suggestion
class MetricsInterfaceTest : public ::testing::Test
{
protected:
    MetricsInterfaceTest() { m_manager = std::make_shared<MetricsManager>(); }

    ~MetricsInterfaceTest() {}

    void TearDown() override {}

    std::shared_ptr<IMetricsManager> m_manager;
};
```